filter_plugins/oo_filters.py

```diff
@@ -229,7 +227,13 @@ def oo_filter_list(data, filter_attr=None):
             raise errors.AnsibleFilterError("|failed expects filter_attr is a str")
 
         # Gather up the values for the list of keys passed in
-        return [x for x in data if x[filter_attr]]
+        return [x for x in data if (filter_attr in x.keys()) and (x[filter_attr])]
```

Comment:
This modifies the behaviour of `oo_filter_list(data, filter_attr)`
If one of the object of `data` misses an attribute named `filter_attr`, it used to crash at runtime. It will now silently ignore that object of `data`.
Which of the two behaviours do we consider as the safest and less error-prone?